src/liquid/components/ld-tooltip/ld-tooltip.tsx

```diff
@@ -0,0 +1,242 @@
+    this.popper.disable()
+    this.visible = false
+  }
+
+  private showTooltip = () => {
+    this.popper.enable()
+    this.visible = true
+  }
+
+  private toggleTooltip = () => {
+    if (this.popper == undefined) {
+      return
+    }
+
+    if (this.visible) {
+      this.hideTooltip()
+    } else {
+      this.showTooltip()
+    }
+  }
+
+  private handleHideTrigger = (event: MouseEvent) => {
+    if (this.triggerType === 'click' || this.disabled) {
+      event.stopPropagation()
+      return
+    }
+
+    clearTimeout(this.delayTimeout)
+
+    if (this.popper) {
+      this.delayTimeout = setTimeout(this.hideTooltip, this.hideDelay)
+    }
+  }
+
+  private handleShowTrigger = (event: MouseEvent) => {
+    if (this.triggerType === 'click' || this.disabled) {
+      event.stopPropagation()
+      return
+    }
+
+    clearTimeout(this.delayTimeout)
+
+    if (this.popper === undefined) {
+      this.delayTimeout = setTimeout(this.initTooltip, this.showDelay)
+    } else {
+      this.delayTimeout = setTimeout(this.showTooltip, this.showDelay)
+    }
+  }
+
+  private handleToggleTrigger = (event: MouseEvent) => {
+    if (this.triggerType === 'hover' || this.disabled) {
+      event.stopPropagation()
+      return
+    }
+
+    if (this.popper === undefined) {
+      this.initTooltip()
+    } else {
+      this.toggleTooltip()
+    }
+  }
+
+  // TODO: maybe this should listen only, if the tooltip was opened by click.
+  @Listen('click', {
+    target: 'window',
+  })
+  handleClickOutside(event) {
+    if (
+      event.target.closest('ld-tooltip') !== this.element &&
+      event.target.closest('.ld-tooltip') !== this.tooltipRef &&
+      this.triggerType === 'click'
+    ) {
+      this.hideTooltip()
+    }
+  }
+
+  // Mobile Safari in some cases does not react to click events on elements
+  // which are not interactive. But it does to touch events.
+  // TODO: maybe this should listen only, if the tooltip was opened by click.
+  @Listen('touchend', {
+    target: 'window',
+    passive: true,
+  })
+  handleTouchOutside(event) {
+    this.handleClickOutside(event)
+  }
+
+  render() {
+    return (
+      <Host>
+        <button
+          aria-describedby={this.id}
+          class={getClassNames([
+            'ld-tooltip__trigger',
+            this.triggerType === 'click' && 'ld-tooltip__trigger--clickable',
+          ])}
+          onClick={this.handleToggleTrigger}
+          onMouseEnter={this.handleShowTrigger}
+          onFocus={this.handleShowTrigger}
+          onMouseLeave={this.handleHideTrigger}
+          onBlur={this.handleHideTrigger}
+          ref={(element) => {
+            this.triggerRef = element
+          }}
+          type="button"
+        >
+          <slot name="trigger">
+            <span class="ld-tooltip__icon">
+              <ld-icon name="info" size="sm" filled />
+            </span>
+          </slot>
+        </button>
+        <div
+          aria-hidden={!this.visible}
```

Comment:
Although setting `aria-hidden` here removes the element from the accessibility tree, the content still remains accessible via keyboard navigation: the user can still tab to the content of the tooltip. You should hide the content for good by setting `visibility: hidden;` on the popper after the transition. You can accomplish this with pure CSS by using a transition which combines opacity and visibility with the visibility transition having a transition duration of 0 and a transition delay of 200ms. This is one approach and there may be others. For more information on this topic, check out this repository: https://github.com/WICG/inert

Reply:
from: { visibility: inherit; }